vnext/Microsoft.ReactNative/IReactPackageBuilderFabric.idl

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import "ViewComponentDescriptor.idl";
+
+#include "DocString.h"
+
+namespace Microsoft.ReactNative
+{
+  [webhosthidden]
+  [experimental]
+  DOC_STRING("Provides ability to register custom ViewComponents when running fabric.  Methods on this interface will eventually be moved onto @IReactPackageProvider once fabric ships.")
+  interface IReactPackageBuilderFabric
```

Comment:
>IReactPackageBuilderFabric [](http://example.com/codeflow?start=12&length=26)

This is the interface that must have the most of the meat. The current design is reversed. This interface has only a single method, while in the ideal world it must have all methods required to define a view component.

Reply:
@vmoroz Do you mean, you want an external `IViewComponent` winrt interface that looks like the internal `IViewComponent` C++ interface? Or like the builder for modules with AddMethod, AddConstants, etc?

Reply:
In the latest code I switched it to use the builder pattern.  The custom component essentially calls into the builder for the methods it wants to override.